Source/Core/Core/IOS/ES/TitleContents.cpp

```diff
@@ -11,43 +11,61 @@
 #include "Common/MsgHandler.h"
 #include "Core/HW/Memmap.h"
 #include "Core/IOS/ES/Formats.h"
+#include "Core/IOS/FS/FileSystemProxy.h"
 #include "Core/IOS/Uids.h"
 
 namespace IOS::HLE
 {
-s32 ESDevice::OpenContent(const ES::TMDReader& tmd, u16 content_index, u32 uid)
+// IPC overhead caused by internal ES <-> FS communications.
+constexpr u64 IPCOverheadTicks = 2700_tbticks;
+
+// Small file read (TMD or content map) overhead. This accounts for the open command,
+// the read (which is assumed to be approximately CLUSTER_READ_TICKS to simplify) and the close.
+constexpr u64 SmallFileReadTicks = 2 * IPCOverheadTicks + 120000_tbticks;
+
+IPCReply ESDevice::OpenContent(const ES::TMDReader& tmd, u16 content_index, u32 uid)
 {
+  u64 ticks = 0;
+
   const u64 title_id = tmd.GetTitleId();
 
   ES::Content content;
   if (!tmd.GetContent(content_index, &content))
-    return ES_EINVAL;
+    return IPCReply{ES_EINVAL};
 
   for (size_t i = 0; i < m_content_table.size(); ++i)
   {
     OpenedContent& entry = m_content_table[i];
     if (entry.m_opened)
       continue;
 
-    auto file = m_ios.GetFS()->OpenFile(PID_KERNEL, PID_KERNEL, GetContentPath(title_id, content),
-                                        FS::Mode::Read);
-    if (!file)
-      return FS::ConvertResult(file.Error());
+    if (content.IsShared())
+      ticks += SmallFileReadTicks;
+
+    const auto [fs_reply, fd] = m_ios.GetFSDevice()->Open(
+        PID_KERNEL, PID_KERNEL, GetContentPath(title_id, content), FS::Mode::Read);
```

Comment:
Idea: wouldn't it be better to make the various calls out to the FS device return the number of ticks they take, instead of needing to compute the number of ticks in ES?